modules/exploits/linux/http/nagios_xi_plugins_check_plugin_authenticated_rce.rb

```diff
@@ -0,0 +1,280 @@
+          }
+      )
+    )
+    register_options [
+      OptString.new('USERNAME', [true, 'Username to authenticate with', 'nagiosadmin']),
+      OptString.new('PASSWORD', [true, 'Password to authenticate with', nil])
+    ]
+  end
+
+  def username
+    datastore['USERNAME']
+  end
+
+  def password
+    datastore['PASSWORD']
+  end
+
+  def finish_install
+    datastore['FINISH_INSTALL']
+  end
+
+  def check
+    # Use nagios_xi_login to try and authenticate. If authentication succeeds, nagios_xi_login returns
+    # an array containing the http response body of a get request to index.php and the session cookies
+    login_result, res_array = nagios_xi_login(username, password, finish_install)
+    case login_result
+    when 1..3 # An error occurred
+      return CheckCode::Unknown(res_array[0])
+    when 4 # Nagios XI is not fully installed
+      install_result = install_nagios_xi(password)
+      if install_result
+        return CheckCode::Unknown(install_result[1])
+      end
+
+      login_result, res_array = login_after_install_or_license(username, password, finish_install)
+      case login_result
+      when 1..3 # An error occurred
+        return CheckCode::Unknown(res_array[0])
+      when 4 # Nagios XI is still not fully installed
+        return CheckCode::Detected('Failed to install Nagios XI on the target.')
+      end
+    end
+
+    # when 5 is excluded from the case statement above to prevent having to use this code block twice.
+    # Including when 5 would require using this code block once at the end of the `when 4` code block above, and once here.
+    if login_result == 5 # the Nagios XI license agreement has not been signed
+      auth_cookies, nsp = res_array
+      sign_license_result = sign_license_agreement(auth_cookies, nsp)
+      if sign_license_result
+        return CheckCode::Unknown(sign_license_result[1])
+      end
+
+      login_result, res_array = login_after_install_or_license(username, password, finish_install)
+      case login_result
+      when 1..3
+        return CheckCode::Unknown(res_array[0])
+      when 5 # the Nagios XI license agreement still has not been signed
+        return CheckCode::Detected('Failed to sign the license agreement.')
+      end
+    end
+
+    print_good('Successfully authenticated to Nagios XI')
+
+    # Obtain the Nagios XI version
+    @auth_cookies = res_array[1] # if we are here, this cannot be nil since the mixin checks for that already
+
+    nagios_version = nagios_xi_version(res_array[0])
+    if nagios_version.nil?
+      return CheckCode::Detected('Unable to obtain the Nagios XI version from the dashboard')
+    end
+
+    print_status("Target is Nagios XI with version #{nagios_version}")
+    # check if the target is actually vulnerable
+    @version = Rex::Version.new(nagios_version)
+    if @version >= Rex::Version.new('5.3.0') && @version <= Rex::Version.new('5.6.5')
+      return CheckCode::Appears
```

Comment:
Wait wait wait wait wait....you said this was from 5.2.0 to 5.6.5 in your module description and the documentation, so why is the check here for 5.3.0 and later and not 5.2.0 and later?

```suggestion
    if @version >= Rex::Version.new('5.2.0') && @version <= Rex::Version.new('5.6.5')
      return CheckCode::Appears
```

Reply:
This has been updated and a fix will be pushed soon. The module will now report all versions less than 5.6.6 as vulnerable as per the original advisory however it will not attempt to exploit targets less than 5.2.0. This keeps in line with the original behavior whilst also allowing users to at least detect older vulnerable instances in their environment so that they are aware of them and know which systems they need to patch.